table/block_based/block_based_table_reader.cc

```diff
@@ -1875,12 +1877,43 @@ CachableEntry<FilterBlockReader> BlockBasedTable::GetFilter(
                    no_io, get_context, lookup_context, prefix_extractor);
 }
 
+void BlockBasedTable::FillBlockCacheAccessRecord(
```

Comment:
You could turn these two into constructors of BlockCacheTraceRecord (and also pass in whatever is needed from rep_, like level). That would also make it easy to add this functionality to the other table readers.

Reply:
Thanks. I removed these two functions and added a constructor for the TraceRecord.